src/app/shared/autocomplete/autocomplete.component.html

```diff
@@ -0,0 +1,118 @@
+             formControlName="city"
+             (click)="toogleDropdown()"
+             autocomplete="off">
+      <div *ngIf="showDropdown" class="city-dropdown-container">
+        <div (click)="changeCity(city)" class="city-text" *ngFor="let city of citiesNames | searchFilter: getCityValue()">
+          {{city}}
+        </div>
+      </div>
+      <span *ngIf="!autocompleteForm.get('city').valid && autocompleteForm.get('city').touched"
+            class="text-muted">
+        Please enter a city!
+      </span>
+    </div>
+
+    <div class="autocomplete-row">
+      <div class="form-group autocomplete-col">
+        <label>Pick-up date</label>
+        <input id="pickUpDate"
+               type="Date"
+               class="form-control"
+               formControlName="pickUpDate">
+        <span *ngIf="!autocompleteForm.get('pickUpDate').valid && autocompleteForm.get('pickUpDate').touched"
+              class="text-muted">
+        Please enter a date!
+      </span>
+      </div>
+
+      <div class="form-group autocomplete-col">
+        <label>Pick-up time</label>
+        <input id="pickUpTime"
+               type="Time"
+               class="form-control"
+               formControlName="pickUpTime">
+        <span *ngIf="!autocompleteForm.get('pickUpTime').valid && autocompleteForm.get('pickUpTime').touched"
+              class="text-muted">
+        Please enter time!
+      </span>
+      </div>
+    </div>
+
+    <div class="autocomplete-row">
+      <div class="form-group autocomplete-col">
+        <label>Drop-off date</label>
+        <input id="dropOffDate"
+               type="Date"
+               class="form-control"
+               formControlName="dropOffDate">
+        <span *ngIf="!autocompleteForm.get('dropOffDate').valid && autocompleteForm.get('dropOffDate').touched"
+              class="text-muted">
+        Please enter a date!
+      </span>
+      </div>
+
+      <div class="form-group autocomplete-col">
+        <label>Drop-off time</label>
+        <input id="dropOffTime"
+               type="Time"
+               class="form-control"
+               formControlName="dropOffTime">
+        <span *ngIf="!autocompleteForm.get('dropOffTime').valid && autocompleteForm.get('dropOffTime').touched"
+              class="text-muted">
+        Please enter time!
+      </span>
+      </div>
+    </div>
+
+    <div class="form-group">
+      <label>Age</label>
+      <select class="custom-select" (change)="changeAgeOption($event)" formControlName="age">
+        <option value="" disabled>Choose your city</option>
+        <option *ngFor="let ageOption of ageOptions" [ngValue]="ageOption">{{ageOption}}</option>
+      </select>
+    </div>
+
+    <button type="submit"
+            class="btn btn-danger search-button"
+            data-toggle="modal"
+            data-target="#resultModal"
+            [disabled]="!autocompleteForm.valid">
+      Search
+    </button>
+  </form>
+
+  <div class="modal fade"
```

Comment:
Modal should be a separate component: modal-widget.component.ts

Reply:
Moved modal to a separate component